packages/block-editor/src/components/off-canvas-editor/block-select-button.js

```diff
@@ -60,6 +61,13 @@ function ListViewBlockSelectButton(
 		}
 	}
 
+	// This is a very annoying kind of exception.
+	// Maybe we need to have a config that allows blocks
+	// to have a different icon in the list view?
+	if ( block.name === 'core/navigation-submenu' ) {
```

Comment:
We should track this in https://github.com/WordPress/gutenberg/issues/46166 if it goes ahead "as is".